src/main/java/ch/njol/skript/entity/WolfData.java

```diff
@@ -18,113 +18,123 @@
+		}
 		return true;
 	}
-	
+
 	@Override
-	public void set(final Wolf entity) {
+	public void set(Wolf entity) {
 		if (angry != 0)
 			entity.setAngry(angry == 1);
 		if (tamed != 0)
 			entity.setTamed(tamed == 1);
-//		if (owner != null) {
-//			if (owner.isEmpty())
-//				entity.setOwner(null);
-//			else
-//				entity.setOwner(Bukkit.getOfflinePlayer(owner));
-//		}
+		if (collarColor != null)
+			entity.setCollarColor(collarColor);
 	}
-	
+
 	@Override
-	public boolean match(final Wolf entity) {
-		return (angry == 0 || entity.isAngry() == (angry == 1)) && (tamed == 0 || entity.isTamed() == (tamed == 1));
-//				&& (owner == null || owner.isEmpty() && entity.getOwner() == null || entity.getOwner() != null && entity.getOwner().getName().equalsIgnoreCase(owner));
+	public boolean match(Wolf entity) {
+		return (angry == 0 || entity.isAngry() == (angry == 1)) && (tamed == 0 || entity.isTamed() == (tamed == 1)) && (collarColor == null ? true : entity.getCollarColor() == collarColor);
 	}
-	
+
 	@Override
 	public Class<Wolf> getType() {
 		return Wolf.class;
 	}
-	
+
 	@Override
 	protected int hashCode_i() {
-		final int prime = 31;
-		int result = 1;
+		int prime = 31, result = 1;
 		result = prime * result + angry;
 		result = prime * result + tamed;
+		result = prime * result + (collarColor == null ? 0 : collarColor.hashCode());
 		return result;
 	}
-	
+
 	@Override
-	protected boolean equals_i(final EntityData<?> obj) {
+	protected boolean equals_i(EntityData<?> obj) {
 		if (!(obj instanceof WolfData))
 			return false;
-		final WolfData other = (WolfData) obj;
+		WolfData other = (WolfData) obj;
 		if (angry != other.angry)
 			return false;
 		if (tamed != other.tamed)
 			return false;
+		if (collarColor != other.collarColor)
+			return false;
 		return true;
 	}
-	
-//		return angry + "|" + tamed;
+
+	/**
+	 * Note that this method is only used when changing Skript versions 2.1 to anything above.
+	 */
+	@Deprecated
 	@Override
-	protected boolean deserialize(final String s) {
-		final String[] split = s.split("\\|");
+	protected boolean deserialize(String s) {
+		String[] split = s.split("\\|");
 		if (split.length != 2)
 			return false;
 		try {
 			angry = Integer.parseInt(split[0]);
 			tamed = Integer.parseInt(split[1]);
 			return true;
-		} catch (final NumberFormatException e) {
+		} catch (NumberFormatException e) {
 			return false;
 		}
 	}
-	
+
 	@Override
-	public boolean isSupertypeOf(final EntityData<?> e) {
-		if (e instanceof WolfData)
-			return (angry == 0 || ((WolfData) e).angry == angry) && (tamed == 0 || ((WolfData) e).tamed == tamed);
+	public boolean isSupertypeOf(EntityData<?> entityData) {
+		if (entityData instanceof WolfData) {
+			WolfData wolfData = (WolfData) entityData;
+			return (angry == 0 || wolfData.angry == angry) && (tamed == 0 || wolfData.tamed == tamed) && (collarColor == null && wolfData.collarColor == null ? true : wolfData.collarColor == collarColor);
```

Comment:
This can be simplified, right?
```suggestion
			return (angry == 0 || wolfData.angry == angry) && (tamed == 0 || wolfData.tamed == tamed) && (wolfData.collarColor == collarColor);
```